### PR TITLE
fix(FEC-10773): reuse Shaka load option to start load from the detached point

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Issue: attachMediaSource start to load from the beginning instead of the last detach point.
Solution: start load from the last detach point.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/dash-adapter.js [hunk 1]
```diff
@@ -553,7 +539,11 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
    */
   detachMediaSource(): void {
     if (this._shaka) {
-      this._lastTimeDetach = this.currentTime;
+      if (parseInt(this.currentTime) === parseInt(this.duration)) {
```
Comment (RoyBregman) [inline src/dash-adapter.js:542]: why do we need to parseInt? They are both numbers, no?

Comment (Yuvalke) [reply src/dash-adapter.js:542]: Right but they're floats could use also Math.round instead

Comment (RoyBregman) [reply src/dash-adapter.js:542]: Yes, i think it is better. ParseInt expects string.
Did it run flow?

Comment (RoyBregman) [reply src/dash-adapter.js:542]: I see that it passed flow. Anyway it is more clearer to math round. Someone might think like me it is was done by mistake

Comment (Yuvalke) [reply src/dash-adapter.js:542]: Sure, it was also in use before - take a look on the line 543 that removed

Comment (Yuvalke) [reply src/dash-adapter.js:542]: Sounds reasonable then

Comment (Yuvalke) [reply src/dash-adapter.js:542]: Actually with round will take 5.5 to 6 and parseInt to 5 but I'll use floor instead to make it clear

Comment (RoyBregman) [reply src/dash-adapter.js:542]: Yes, I meant Math.Floor of course

Comment (Yuvalke) [reply src/dash-adapter.js]: Super edge case that I see now is 800.1 and 799.9 which return false so I added handling for this as well

Comment (RoyBregman) [reply src/dash-adapter.js]: The worst case scenario is that the user will see the end of the playback and click replay, no?

Comment (RoyBregman) [reply src/dash-adapter.js]: So Actually Math.round is better, no?

Comment (Yuvalke) [reply src/dash-adapter.js]: Same for this solution

File: src/dash-adapter.js [hunk 2]
```diff
@@ -553,7 +539,12 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
    */
   detachMediaSource(): void {
     if (this._shaka) {
-      this._lastTimeDetach = this.currentTime;
+      // 1 second from will signal as end
```
Comment (OrenMe) [inline src/dash-adapter.js:542]: 1 sec from? Please add details

File: src/dash-adapter.js (hunk 1 above)
Comment (Yuvalke) [reply src/dash-adapter.js]: I wonder if we need the floor or keep it simple

**Review by RoyBregman — DISMISSED** [review]

File: src/dash-adapter.js [hunk 3]
```diff
@@ -553,7 +539,12 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
    */
   detachMediaSource(): void {
     if (this._shaka) {
-      this._lastTimeDetach = this.currentTime;
+      // 1 second different between duration and current time will signal as end - will enable replay button
+      if (Math.floor(this.duration - this.currentTime) === 0) {
+        this._lastTimeDetach = 0;
+      } else {
+        this._lastTimeDetach = this.currentTime;
+      }
```
Comment (yairans) [inline src/dash-adapter.js:547]: this overrides the startPosition passed by the app in case of pre-roll

**Review by yairans — CHANGES_REQUESTED** [review]

**Review by yairans — APPROVED** [review]

**Pull request merged**